rust-sdk/core/src/math/bundle.rs

```diff
@@ -0,0 +1,117 @@
+#[cfg(feature = "wasm")]
+use wasm_bindgen::prelude::*;
+
+use ethnum::U256;
+
+use crate::POSITION_BUNDLE_SIZE;
+
+const POSITION_BUNDLE_BYTES: usize = POSITION_BUNDLE_SIZE / 8;
+
+/// Get the first unoccupied position in a bundle
+///
+/// # Arguments
+/// * `bundle` - The bundle to check
+///
+/// # Returns
+/// * `u32` - The first unoccupied position
+#[cfg_attr(feature = "wasm", wasm_bindgen(js_name = firstUnoccupiedPositionInBundle, skip_jsdoc))]
+pub fn first_unoccupied_position_in_bundle(bitmap: &[u8]) -> u32 {
+    let value = bitmap_to_u256(bitmap);
+    for i in 0..POSITION_BUNDLE_SIZE {
+        if value & (U256::ONE << i) == 0 {
+            return i as u32;
+        }
+    }
+    panic!("No unoccupied position in bundle");
```

Comment:
Wouldn't know what happens in js, I guess throws an exception. Panic is also suboptimal since it might bloat the wasm bundle. Will refactor this to return something meaningful in this case

Reply:
Now it uses `Option`👍️